src/vs/workbench/browser/parts/editor/tabsTitleControl.ts

```diff
@@ -1292,6 +1292,22 @@ registerThemingParticipant((theme: ITheme, collector: ICssStyleCollector) => {
 		if (tabActiveBackground && adjustedTabBackground && adjustedTabDragBackground) {
 			const adjustedColor = tabActiveBackground.flatten(adjustedTabBackground);
 			const adjustedColorDrag = tabActiveBackground.flatten(adjustedTabDragBackground);
+			collector.addRule(`
+				.monaco-workbench .part.editor > .content:not(.dragged-over) .editor-group-container.active > .title .tabs-container > .tab.sizing-shrink.active:not(.dragged) > .tab-label::after {
+					background: linear-gradient(to left, ${adjustedColor}, transparent);
+				}
+
+				.monaco-workbench .part.editor > .content.dragged-over .editor-group-container.active > .title .tabs-container > .tab.sizing-shrink.active:not(.dragged) > .tab-label::after {
+					background: linear-gradient(to left, ${adjustedColorDrag}, transparent);
+				}
+			`);
+		}
+
+		// Adjust gradient for unfocused active tab background
+		const tabUnfocusedActiveBackground = theme.getColor(TAB_UNFOCUSED_ACTIVE_BACKGROUND);
+		if (tabUnfocusedActiveBackground && adjustedTabBackground && adjustedTabDragBackground) {
+			const adjustedColor = tabUnfocusedActiveBackground.flatten(adjustedTabBackground);
+			const adjustedColorDrag = tabUnfocusedActiveBackground.flatten(adjustedTabDragBackground);
 			collector.addRule(`
 				.monaco-workbench .part.editor > .content:not(.dragged-over) .editor-group-container > .title .tabs-container > .tab.sizing-shrink.active:not(.dragged) > .tab-label::after {
```

Comment:
@dmolin I just realised that for this specific rule we need to use `.monaco-workbench .part.editor > .content:not(.dragged-over) .editor-group-container:not(.active)` (note the extra `:not(.active)`) to ensure we only define it for inactive groups and not overwrite the style for active groups. Because it could be that no color for active group is defined and then this rule would still apply.

Reply:
you're right, good catch! :+1:

Reply:
Wait.. is this a thing? I mean, can it actually happen? from what I see, both colors are always defined: if not present in any `settings.json`, they simply get the same color (taken from the editor background color). In that case, with the rewrite that applies the CSS rule only once, they would fall into the "common" rule case (the one that doesn't even add ".active" to the css selector) and the ".active" rule would never be defined in that case

Reply:
@dmolin yeah in this case you are right the color will always be defined

Reply:
see latest commit f763adc